packages/core/src/PlayerContextProvider.js

```diff
@@ -495,21 +498,24 @@ export class PlayerContextProvider extends Component {
       });
   }
 
-  setMediaElementSources(sources) {
+  setMediaElementSources(playlist, trackIndex) {
     // remove current sources
     let firstChild;
     while ((firstChild = this.media.firstChild)) {
       this.media.removeChild(firstChild);
     }
-    // add new sources
-    for (const source of sources) {
-      const sourceElement = document.createElement('source');
-      sourceElement.src = source.src;
-      if (source.type) {
-        sourceElement.type = source.type;
+    if (isPlaylistValid(playlist)) {
```

Comment:
I think it makes enough sense to just look up `this.props.playlist` and `this.state.activeTrackindex` here instead of requiring them as arguments.

Reply:
Sure